kyo-core/shared/src/test/scala/kyoTest/layersTest.scala

```diff
@@ -0,0 +1,85 @@
+package kyoTest
+
+import kyo.*
+// import kyo.syntax.*
+
+// TODO:
+// - Replace Envs with the implementation of EnvMaps
+// - Allow Envs.run to accept a layer...
+// - kyo.provide(food, heat, stomach, spells)
+// - kyo.provide(food, heat, stomach, spells)
+// - Envs.run(1)
+class layersTest extends KyoTest:
```

Comment:
Do we need to tests for cyclic dependencies? Or is that not possible with the encoding?

Reply:
I don't think that makes much sense. I did add a test for when `In =:= Out`, which we don't want to break memoization.